linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,8 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (hash.ContainsKey (tr))
 					continue;
+				if (!CanChangeScope (tr))
```

Comment:
Hmm, why do you want to have `CanChangeScope` in one of the loops changing the scope only?